code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/notification/PushTemplateConstants.kt

```diff
@@ -0,0 +1,164 @@
+/*
+  Copyright 2024 Adobe. All rights reserved.
+  This file is licensed to you under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License. You may obtain a copy
+  of the License at http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software distributed under
+  the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+  OF ANY KIND, either express or implied. See the License for the specific language
+  governing permissions and limitations under the License.
+*/
+
+package com.adobe.marketing.mobile.services.ui.notification
+
+/**
+ * This object holds all constant values for handling out-of-the-box push template notifications
+ */
+internal object PushTemplateConstants {
+    const val LOG_TAG = "AEPSDK Push Templates"
+    const val CACHE_BASE_DIR = "pushtemplates"
+    const val PUSH_IMAGE_CACHE = "pushimagecache"
+    const val DEFAULT_CHANNEL_ID = "AEPSDKPushChannel"
+
+    // When no channel name is received from the push notification, this default channel name is
+    // used.
+    // This will appear in the notification settings for the app.
+    const val DEFAULT_CHANNEL_NAME = "AEPSDK Push Notifications"
+    const val SILENT_CHANNEL_NAME = "AEPSDK Silent Push Notifications"
+
+    internal object NotificationAction {
+        const val DISMISSED = "Notification Dismissed"
+        const val OPENED = "Notification Opened"
+        const val BUTTON_CLICKED = "Notification Button Clicked"
+    }
+
+    internal class Tracking private constructor() {
+        internal object Keys {
+            const val ACTION_ID = "actionId"
+            const val ACTION_URI = "actionUri"
+        }
+    }
+
+    internal object DefaultValues {
+        const val SILENT_NOTIFICATION_CHANNEL_ID = "AEPSDK Silent Push Notifications"
+        const val SILENT_SOUND = "silent"
+        const val CAROUSEL_MAX_BITMAP_WIDTH = 300
+        const val CAROUSEL_MAX_BITMAP_HEIGHT = 200
+        const val AUTO_CAROUSEL_MODE = "auto"
+        const val DEFAULT_MANUAL_CAROUSEL_MODE = "default"
+        const val FILMSTRIP_CAROUSEL_MODE = "filmstrip"
+        const val CAROUSEL_MINIMUM_IMAGE_COUNT = 3
+        const val MANUAL_CAROUSEL_START_INDEX = 0
+        const val FILMSTRIP_CAROUSEL_CENTER_INDEX = 1
+        const val NO_CENTER_INDEX_SET = -1
+
+        // TODO: revisit this value. should cache time be configurable rather than have a static
+        // value?
+        const val PUSH_NOTIFICATION_IMAGE_CACHE_EXPIRY_IN_MILLISECONDS: Long = 259200000 // 3 days
```

Comment:
- `const val PUSH_NOTIFICATION_IMAGE_CACHE_EXPIRY_IN_MILLISECONDS = TimeUnit.DAYS.toMillis(X)`
- Can we check on what we do on iOS with caching and how long are we caching there?

Reply:
`const val PUSH_NOTIFICATION_IMAGE_CACHE_EXPIRY_IN_MILLISECONDS = TimeUnit.DAYS.toMillis(X)` is not usable as the compiler complains that the value should be constant:
`Const 'val' initializer should be a constant value`

Reply:
Converted to `val` instead of `const val`